### PR TITLE
Various minore fixes in the amdgpu plugin

### PR DESCRIPTION
**Pull request by avagin**

### CODE REVIEW

File: plugins/amdgpu/amdgpu_plugin.c [hunk 1]
```diff
@@ -165,6 +165,7 @@ FILE *open_img_file(char *path, bool write, size_t *size)
 	fp = fdopen(fd, write ? "w" : "r");
```
Comment (rst0git) [inline plugins/amdgpu/amdgpu_plugin.c:165]: there is a typo in the commit message (*_error path_)

**Review by dayatsin-amd — APPROVED** [review]

**Review by dayatsin-amd — APPROVED** [review]

**Review by dayatsin-amd — COMMENTED** [review]
All 3 patches look good to me.
@fdavid-amd Can you test these patches

File: plugins/amdgpu/amdgpu_plugin.c [hunk 2]
```diff
@@ -1795,7 +1796,12 @@ int amdgpu_plugin_restore_file(int id)
 		 * copy of the fd. CRIU core owns the duplicated returned fd, and amdgpu_plugin owns the fd stored in
 		 * tp_node.
 		 */
-		return dup(fd);
+		fd = dup(fd);
+		if (fd == -1) {
+			pr_perror("unable to duplicate the render fd");
+			return -1
```
Comment (rst0git) [inline plugins/amdgpu/amdgpu_plugin.c:1802]: missing a semicolon here

**Comment by rst0git** [comment]
I tested these changes locally with AMD MI100, and I can confirm that the checkpoint/restore works as expected.

File: plugins/amdgpu/amdgpu_plugin.c (hunk 2 above)
Comment (rst0git) [reply plugins/amdgpu/amdgpu_plugin.c:1802]: I've opened a pull request to build the amdgpu plugin in CI that should catch similar errors: https://github.com/checkpoint-restore/criu/pull/2276

**Review by rst0git — APPROVED** [review]
LGTM

**Pull request merged**